### PR TITLE
ActionParams

### PR DESCRIPTION
**Pull request by tomer-epstein**

### CODE REVIEW

File: backend/src/types.ts [hunk 1]
```diff
@@ -46,3 +52,33 @@ export interface IImage {
...
+    },
+    action2: {
+        name: "",
+        action: action,
+        contexts: [{
+            project: "",
+            params: 3
+        }]
+    }
+}
```
Comment (perspectivus1) [inline backend/src/types.ts:84]: This is an example and should be removed

File: bas-platform/src/actions/impl.ts [hunk 2]
```diff
@@ -33,19 +34,19 @@ export class CommandAction extends Action implements ICommandAction {
...
         this.contributorId = "";
-        this.context = "";
         this.snippetName = "";
+        this.context = "";
     }
 }
 
 export class FileAction extends Action implements IFileAction {
-    uri: Uri;
+    uri: FileActionParams;
```
Comment (perspectivus1) [inline bas-platform/src/actions/impl.ts:49]: It seems that `uri: vscode.Uri` is much clearer. I'm not sure we need to make changes to `bas-platform` at all...

**Review by perspectivus1 — CHANGES_REQUESTED** [review]

File: backend/src/types.ts (hunk 1 above)
Comment (tomer-epstein) [reply backend/src/types.ts:84]: Of course. It's temporary here.

File: bas-platform/src/actions/impl.ts (hunk 2 above)
Comment (tomer-epstein) [reply bas-platform/src/actions/impl.ts:49]: But then what if bas-platform api will chabge then we'll have to change guided-dev api.

Comment (perspectivus1) [reply bas-platform/src/actions/impl.ts:49]: > But then what if bas-platform api will chabge then we'll have to change guided-dev api.

True. I'll approve the PR. Just remove the sample code from `types.ts`

**Comment by perspectivus1** [comment]
Go ahead an publish to npm.

When done, make adjustments to the code, so `backend` compiles and the tests pass.

Then I'll approve the PR...

**Comment by tomer-epstein** [comment]
Usage:

`
import { Uri } from "vscode";

let action: IFileAction = {

    uri: undefined,
    actionType: ActionType.File
}
let item: IItem = {

    id: "",
    title: "",
    description : "",
    labels: [],
    action1: {
        name: "",
        action: {
            actionType: ActionType.Execute,
            executeAction: () => { return Promise.resolve();}
        },
        contexts: [{
            project: "",
            params: ["param_1", "param_2"]
        }]
    },
    action2: {
        name: "",
        action: action,
        contexts: [{
            project: "",
            uri: Uri.parse("")
        }]
    }
}
`

**Review by perspectivus1 — COMMENTED** [review]
No changes in code to vscode-contrib1, 2 and 3?

**Review by perspectivus1 — APPROVED** [review]

**Pull request merged**